src/NHibernate.Test/SystemTransactions/SystemTransactionFixture.cs

```diff
@@ -459,4 +461,18 @@ public class SystemTransactionWithoutConnectionFromPrepareFixture : SystemTransa
 	{
 		protected override bool UseConnectionOnSystemTransactionPrepare => false;
 	}
+
+	public class SystemTransactionWithoutConnectionAutoEnlistmentFixture : SystemTransactionFixture
```

Comment:
This PR has ceased relying on connection auto-enlistment, but I had not put any test for ensuring it was actually the case. Now done for non distributed tests. I do not expect distributed to change anything about that, so I have not added a distributed version of those tests. (Connection enlistment with an active transaction does not occur during concurrent events like transaction completion or second phase callbacks.)